working/0041 - static extension methods/lrn-static-extension-survey.md

```diff
@@ -0,0 +1,177 @@
+# Scoped Static Extension Methods Survey
+
+[lrn@google.com](mailto:lrn@google.com)
+
+
+# Background
+
+Dart 2 has a design that makes it very hard to change an API, and while `.`-chain notation is convenient, it breaks down if you have to call static methods instead of instance methods.
+
+_Scoped static extension methods_ is a language feature which introduces static methods that can be called as if they were instance methods (improved syntax) based on the static type of the receiver expression (static dispatch). The methods are only in scope if their declaration is (scoped).
+
+There is a number of design choices that are not given from the above, and which have to be resolved before we can claim to have a full design for Dart.
+
+We will look to other languages for inspiration, but Dart is fairly unique in having reified type arguments and _not_ having overloading on method signature. Solutions that work for other languages may not fit Dart as well.
+
+
+# Other Languages
+
+A number of other languages have something that can be described as static extension methods.
+
+
+## C#
```

Comment:
Might be worth pointing out that C# is working on adding support for extension getters, static methods, etc.: https://github.com/dotnet/roslyn/issues/11159